### PR TITLE
Improved README.md, separate effect and key shortcut lists, table of contents

### PR DESCRIPTION
**Pull request by freq-mod**
#297 Draft, because it will likely need polishing and adding information about Windows and MacOS.

Build info about GNU/Linux OS's have been updated to reflect latests changes, some typos and awkward grammar were corrected.

### CODE REVIEW

**Review by OPNA2608 — COMMENTED** [review]
Made some suggestions. What's the scope of this, do you want to address #297 or just fix your specific compilation problems?

File: README.md [hunk 1]
```diff
@@ -310,25 +162,31 @@ apt install \
...
 ```
 
-##### FreeBSD
-You may skip to "Compilation" - "Linux / BSD" - "FreeBSD" if you want to build via FreeBSD ports.
+##### Arch Linux/Manjaro/Artix:
+**TODO** 
 
-### Compilation
-Generally, BambooTracker releases are built like this:
+- check out AUR package and request fixing it if needed
```
Comment (OPNA2608) [inline README.md:168]: I believe the AUR packaging as been broken and outdated for awhile but I don't use Arch (btw) and didn't see the point in registering on the AUR just to comment on its brokenness.

File: README.md [hunk 2]
```diff
@@ -343,16 +201,33 @@ If you know that building with an old version of JACK that lacks the `jack_port_
...
+```
+git submodule init
+git submodule update
+```
+... else specify which Rt libraries should be looked up with pkg-config
+`export BT_OPTIONS="$BT_OPTIONS CONFIG+=system_rtaudio CONFIG+=system_rtmidi"`
 
-- **PulseAudio Support**: `CONFIG+=use_pulse`
-- **JACK Support**: `CONFIG+=use_jack CONFIG+=jack_has_rename`
+If you want to build with any of the optional dependencies, append the following options to `export`:
```
[the remark this answers is not in the document]
Comment (freq-mod) [reply README.md:218]: Thanks for the info. I will mention it.

File: README.md (hunk 1 above)
Comment (freq-mod) [reply README.md:168]: Perhaps, I can remove it for time being but still I would like to have mention on working AUR package for easier installation on btwOS-based distros

**Comment by freq-mod** [comment]
>What's the scope of this, do you want to address #297 or just fix your specific compilation problems?

Both. While I finally got BambooTracker to compile, at some point somebody else will run into same problems. And I think, what was outlined in #297, still holds true.

File: README.md [hunk 3]
```diff
@@ -310,25 +162,27 @@ apt install \
   libjack-jackd2-dev # for JACK2, or libjack-dev for JACK1
 ```
 
+##### Arch Linux/Manjaro/Artix:
+```bash
+pacman -S \
+qt5-tools \
+pulseaudio \
+jack \ 
+alsa-plugins #uncertain about this one though
```
Comment (OPNA2608) [inline README.md:171]: Nitpick but indentation would be nicer imo.

```suggestion
pacman -S \
  qt5-tools \
  pulseaudio \
  jack \ 
  alsa-plugins #uncertain about this one though
```

**Comment by OPNA2608** [comment]
Can you update the dependency instructions for Debian/Ubuntu? [QtMultimedia was dropped there as well](https://github.com/rerrahkr/BambooTracker/blob/5bf3fd404d1c340a2051eb56632b177a971f26ca/.github/workflows/ubuntu.yml#L56-L60).

Ideally the dependencies & compile instructions should prefer using your system's own Rt-libraries in favour of a vendored one, because Debian/Arch/Nixpkgs/etc's version of them may be patched to fix system-specific problems. I sadly don't know what release version of RtAudio/RtMidi we require though, for example Ubuntu 16.04's versions of them didn't worked when i tested it on CI. I can confirm that the latest releases (RtAudio 5.1.0 and RtMidi 4.0.0) work.

**Comment by freq-mod** [comment]
@OPNA2608 
>Can you update the dependency instructions for Debian/Ubuntu as well? QtMultimedia was dropped there as well.

Done, thanks for the info

>I can confirm that the latest releases (RtAudio 5.1.0 and RtMidi 4.0.0) work.

Ubuntu 20.04 has RtAudio 5.0.0 and RtMidi 3.0.0. I might check to see if they work

File: README.md (hunk 1 above)
Comment (freq-mod) [reply README.md]: @jpcima, iirc you are a maintainer of AUR package of BT, would you mind fixing and updating it?

File: README.md [hunk 4]
```diff
@@ -304,31 +170,37 @@ The following optional dependencies exist:
...
 
+##### Arch Linux/Manjaro/Artix:
+```bash
+pacman -S \
+  qt5-tools \
+  pulseaudio \
+  jack \ 
+  rtaudio \ #optionally
+  rtmidi \ #optionally
+  alsa-plugins #uncertain about this one though
```
Comment (OPNA2608) [inline README.md:189]: Is this something you wanted to verify still?

**Comment by OPNA2608** [comment]
I'd like it if the minimum known-working versions of RtAudio and RtMidi are listed somewhere in the README, now that we have determined them to be 5.1.0 and 4.0.0 respectively.

File: README.md (hunk 4 above)
Comment (freq-mod) [reply README.md]: Hmm.. must verify this as well: https://aur.archlinux.org/packages/bambootracker-git/

Comment (freq-mod) [reply README.md]: #311 :thinking:

**Comment by freq-mod** [comment]
OK, is there anything else that can be added, improved or corrected? Or do you think it's ready to merge?

**Comment by rerrahkr** [comment]
LGTM. It is also necessary to fix CI deployment, but I think it's okay to do it after merging.

**Comment by freq-mod** [comment]
Then please feel free to merge this.

**Comment by OPNA2608** [comment]
In scripts/package_windows.sh and scripts/package_osx.sh, change
`cp -va ../{img,*.md} .`
to
`cp -va ../*.md .`
to hopefully fix those packaging errors. Sorry for that :sweat:.

**Review by OPNA2608 — APPROVED** [review]
LGTM. Will need to be ported to README_ja.md, I hope Rerrah doesn't mind all the text changes that will need to be retranslated.

**Comment by rerrahkr** [comment]
Thanks for your contributions!

**Pull request merged**